stdlib/public/Backtracing/Backtrace.swift

```diff
@@ -0,0 +1,491 @@
+//===--- Backtrace.swift --------------------------------------*- swift -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+//  Defines the `Backtrace` struct that represents a captured backtrace.
+//
+//===----------------------------------------------------------------------===//
+
+import Swift
+
+#if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
+
+@_implementationOnly import Darwin.Mach
+@_implementationOnly import _SwiftBacktracingShims
+
+#endif
+
+/// Holds a backtrace.
+public struct Backtrace: CustomStringConvertible, Sendable {
+  /// The type of an address.
+  ///
+  /// This will be `UInt32` or `UInt64` on current platforms.
+  #if arch(x86_64) || arch(arm64)
+  public typealias Address = UInt64
+  #elseif arch(i386) || arch(arm)
+  public typealias Address = UInt32
+  #else
+  #error("You need to fill this in for your architecture.")
+  #endif
+
+  /// The unwind algorithm to use.
+  public enum UnwindAlgorithm {
+    /// Choose the most appropriate for the platform.
+    case auto
+
+    /// Use the fastest viable method.
+    ///
+    /// Typically this means walking the frame pointers.
+    case fast
+
+    /// Use the most precise available method.
+    ///
+    /// On Darwin and on ELF platforms, this will use EH unwind
+    /// information.  On Windows, it will use Win32 API functions.
+    case precise
+  }
+
+  /// Represents an individual frame in a backtrace.
+  public enum Frame: CustomStringConvertible, Sendable {
+    /// An accurate program counter.
```

Comment:
"Accurate" means that it points to the instruction responsible; the alternative is "return address", which instead points to the instruction *after* the subroutine call, which means that when symbolicating we need to adjust the value (otherwise we'll get the backtrace wrong in some cases). I'll add more explanation in the comment.

Reply:
I've updated the comment. On reflection I decided to remove "accurate" and add more explanation below.